src/write.c

```diff
@@ -562,11 +562,12 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         return AVIF_RESULT_UNSUPPORTED_DEPTH;
     }
 
-    if (!firstCell->width || !firstCell->height) {
-        return AVIF_RESULT_NO_CONTENT;
-    }
-
-    if ((cellCount > 1) && ((firstCell->width < 64) || (firstCell->height < 64))) {
+    if ((cellCount > 1) && !avifAreGridDimensionsValid(firstCell->yuvFormat,
```

Comment:
IMPORTANT: This change loses the `if (!firstCell->width || !firstCell->height)` check in the original code when `cellCount` is 1.

Perhaps we should move the `if ((imageW == 0) || (imageH == 0)` check out of `avifAreGridDimensionsValid()` and resurrect the `if (!firstCell->width || !firstCell->height)` check here.